### PR TITLE
MapCatalog and MapTemplates plugins doc

### PR DESCRIPTION
**Pull request by vlt1**
## Description
Added a little documentation for MapCatalog and MapTemplates plugins.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: docs

## Issue
[mapstore2-georchestra/#199](https://github.com/georchestra/mapstore2-georchestra/issues/199)

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/plugins/MapCatalog.jsx [hunk 1]
```diff
@@ -29,6 +29,20 @@ import { createPlugin } from '../utils/PluginsUtils';
...
+ * @class
+ * @name MapCatalog
+ * @prop {boolean} [active] true if the panel is active
+ * @prop {string} [mapType] current map viewer type
+ * @prop {object} [user] user information
+ * @prop {boolean} [triggerReloadValue] boolean value used to force the reload of the maps list
+ * @prop {function} [onToggleControl] toggle map templates panel visibility callback
+ * @prop {function} [onTriggerReload] callback that changes triggerReloadValue to force the reload of the maps list
+ * @prop {function} [onDelete] triggers map deletion
+ * @prop {function} [onSave] triggers map saving
```
Comment (offtherailz) [inline web/client/plugins/MapCatalog.jsx]:
```suggestion
```
Please jsdoc (for plugin) only configurable properties.

File: web/client/plugins/MapCatalog.jsx [hunk 2]
```diff
@@ -29,6 +29,20 @@ import { createPlugin } from '../utils/PluginsUtils';
 import mapcatalog from '../reducers/mapcatalog';
 import * as epics from '../epics/mapcatalog';
 
+/**
+ * Allows users to browse/edit/remove existing maps
```
Comment (offtherailz) [inline web/client/plugins/MapCatalog.jsx]:
```suggestion
 * Allows users to browse existing maps directly on the map. 
 * Users can search and browse (open, edit, delete, share) maps from a panel available in the BurgerMenu.
```

File: web/client/plugins/MapCatalog.jsx [hunk 3]
```diff
@@ -67,7 +81,7 @@ const MapCatalogComponent = ({
 
 export default createPlugin('MapCatalog', {
     component: connect(createStructuredSelector({
-        active: state => state.controls?.mapCatalog?.enabled,
+        active: state => state.controls && state.controls.mapCatalog && state.controls.mapCatalog.enabled,
```
Comment (offtherailz) [inline web/client/plugins/MapCatalog.jsx]: This is ok, but why did you removed `?.` notation ? Did you find some problems? We updated babel to translate it.
I noticed it becomes a little unreadable in debug, maybe for this reason?

Comment (vlt1) [reply web/client/plugins/MapCatalog.jsx]: After running npm run doctest I got jsdoc parsing error "Unexpected token"

**Pull request merged**